lib/reverse_sentence.rb

```diff
@@ -1,6 +1,32 @@
 # A method to reverse the words in a sentence, in place.
-# Time complexity: ?
-# Space complexity: ?
+# Time complexity: O(n) - linear complexity, will scale according to the growth of the array length
+# Space complexity: O(n) - linear complexity, an array is being created
 def reverse_sentence(my_sentence)
-  raise NotImplementedError
+  return my_sentence if my_sentence.nil? || my_sentence.length <= 1
+  
+  sentence = my_sentence.split(/(\S+|\s{2,})/)
```

Comment:
So here you're splitting the string into an array of words.  So this is **not** in place.